lib/tasks/comment_tasks.rake

```diff
@@ -0,0 +1,40 @@
+namespace :comment do
+  desc "Syncs the Comments.approved column to the new Comments.spam column"
+  task(:sync_approved_to_spam, [:limit] => :environment) do |_t, args|
+    # DB admin can pass in an optional integer for the limit
+    limit = args&.limit&.to_i
+
+    # success_count is used to provide status updates as the rake task is running.
+    success_count = 0
+
+    # Count is only used when a limit is provided. Otherwise, it is ignored.
+    count = 0 if limit.present?
+
+    # The approved and spam columns are inverse values of each other. If approved is true, then spam
+    # should be false, and vice versa. Thus, only records where approved and spam values are the same need to be synced.
+    Comment.where("(approved IS TRUE AND spam IS TRUE) OR (approved IS FALSE AND spam IS FALSE)").find_in_batches do |batch|
+      batch.each do |comment|
+        if comment.approved == comment.spam # Guard clause: guards against
+          # unnecessary updates to the db if the query returns comments that have already been synced.
+          comment.update_attribute(:spam, !comment.approved)
+          success_count += 1
+        end
+
+        next if limit.blank?
+
+        # Provide verbose logging if limit is set
+        puts "Updated comment #{comment.id} spam attribute to #{comment.spam}."
+        # Break out of loop if limit is met or exceeded
+        count += 1
+        break if limit <= count
+      end
+      # Provide periodic logging to monitor rake task progress.
+      puts "#{success_count} comments synced."
```

Comment:
You can find examples in the after tasks for how we usually do progress updates like this, look for `find_in_batches.with_index` (though note that we do not want to get the total count of comment batches to update, because that count query would also kill the db)